frontend/apps/desktop/src/utils/navigation-container.tsx

```diff
@@ -48,6 +48,12 @@ export function NavigationContainer({
     return navigation.state.subscribe(() => {
       const state = navigation.state.get()
       send('windowNavState', state)
+      // Emit route change event to main process
+      send('routeChange', {
```

Comment:
I don't see why this is necessary. The middle end already has the navigation state at this point, sent on line 50. So why do we send all this stuff again in a separate event?

Reply:
I can remove it if you are confident this is being sent before.

Reply:
yes I'm confident because the whole state is sent right here on the previous line `send('windowNavState', state)`